cdap-app-templates/cdap-program-report/src/main/java/co/cask/cdap/report/ReportGenerationSpark.java

```diff
@@ -154,10 +159,14 @@ public void getReportStatus(HttpServiceRequest request, HttpServiceResponder res
         return;
       }
       long creationTime = ReportIds.getTime(reportId, TimeUnit.SECONDS);
-      // Read the report request from _START file, which was written at the beginning of report generation
-      String reportRequest =
+      ReportStatus status = getReportStatus(reportIdDir);
+      // read the report request from _START file, which was written at the beginning of report generation
+      String reportRequestString =
         new String(ByteStreams.toByteArray(reportIdDir.append(START_FILE).getInputStream()), StandardCharsets.UTF_8);
-      responder.sendJson(new ReportGenerationInfo(creationTime, getReportStatus(reportIdDir), reportRequest));
+      ReportGenerationRequest reportRequest = GSON.fromJson(reportRequestString, REPORT_GENERATION_REQUEST_TYPE);
+      ReportGenerationInfo reportGenerationInfo = new ReportGenerationInfo(reportRequest.getName(), null, creationTime,
```

Comment:
similar comment about description. also for error we set to `null` ?  will there be a PR to set the error message if reportStatus is failed ?

Reply:
do we store the error information already or does it have to be added ?

Reply:
yea, it already exists. I'll add this too